src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,117 @@
+package mate.academy.dao;
+
+import mate.academy.excpetion.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+import mate.academy.util.ConnectionUtil;
+
+import java.sql.*;
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    private static final int ID_PART = 3;
+    private static final int TITLE_PART = 1;
+    private static final int PRICE_PART = 2;
+    private static final int MINIMUM_CHANGES = 1;
+    private static final int INDEX_ID = 1;
+    private static final String INSERT_INTO_DB = "INSERT INTO books (title, price) VALUES (?, ?)";
+    private static final String SELECT_GET_BY_ID = "SELECT * FROM books WHERE id = ?";
+    private static final String SELECT_ALL = "SELECT * FROM books";
+    private static final String UPDATE_BOOK = "UPDATE books SET title = ?, price = ? WHERE id = ?";
+    private static final String DELETE_BY_ID = "DELETE FROM books WHERE id = ?";
+
+    @Override
+    public Book create(Book book) {
+        try (Connection connection = ConnectionUtil.getConnection();
+             PreparedStatement statement = connection.prepareStatement(INSERT_INTO_DB,
+                     Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(TITLE_PART, book.getTitle());
+            statement.setBigDecimal(PRICE_PART, book.getPrice());
+            if (statement.executeUpdate() < MINIMUM_CHANGES) {
+                throw new RuntimeException("Can't insert less than one row");
+            }
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                Long id = resultSet.getObject(INDEX_ID, Long.class);
+                book.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't insert book = " + book + " into DB" ,e);
+        }
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection();
+             PreparedStatement statement = connection.prepareStatement(SELECT_GET_BY_ID)) {
+            statement.setLong(INDEX_ID, id);
+            ResultSet resultSet = statement.executeQuery();
+            Book book = new Book();

```

Comment:
You create book 2 times, first time here in 54 row, second time in getBoot method